subsys/net/ip/rpl.c

```diff
@@ -1642,8 +1642,7 @@ static void set_dao_lifetime_timer(struct net_rpl_instance *instance)
 		u32_t expiration_time;
 
 		expiration_time = (u32_t)instance->default_lifetime *
-			(u32_t)instance->lifetime_unit *
-			MSEC_PER_SEC / 2;
+			(u32_t)instance->lifetime_unit * K_SECONDS(1) / 2;
```

Comment:
Maybe K_MSEC(500) after all?

Reply:
The comment few lines above says "Set up another DAO within half the expiration time" so in this respect it makes sense to leave the division in place.